php-checks/src/main/java/org/sonar/php/checks/VarKeywordUsageCheck.java

```diff
@@ -19,34 +19,45 @@
  */
 package org.sonar.php.checks;
 
-import com.sonar.sslr.api.AstNode;
 import org.sonar.api.server.rule.RulesDefinition;
 import org.sonar.check.Priority;
 import org.sonar.check.Rule;
 import org.sonar.php.api.PHPKeyword;
+import org.sonar.plugins.php.api.tree.declaration.ClassPropertyDeclarationTree;
+import org.sonar.plugins.php.api.tree.lexical.SyntaxToken;
+import org.sonar.plugins.php.api.visitors.PHPVisitorCheck;
 import org.sonar.squidbridge.annotations.SqaleConstantRemediation;
 import org.sonar.squidbridge.annotations.SqaleSubCharacteristic;
-import org.sonar.squidbridge.checks.SquidCheck;
-import org.sonar.sslr.parser.LexerlessGrammar;
 
 @Rule(
-  key = "S1765",
+  key = VarKeywordUsageCheck.KEY,
   name = "The \"var\" keyword should not be used",
   priority = Priority.MINOR,
   tags = {Tags.CONVENTION, Tags.PSR2})
 @SqaleSubCharacteristic(RulesDefinition.SubCharacteristics.UNDERSTANDABILITY)
 @SqaleConstantRemediation("2min")
-public class VarKeywordUsageCheck extends SquidCheck<LexerlessGrammar> {
+public class VarKeywordUsageCheck extends PHPVisitorCheck {
+
+  public static final String KEY = "S1765";
+  private static final String MESSAGE = "Replace the \"var\" keyword with the modifier \"public\".";
 
   @Override
-  public void init() {
-    subscribeTo(
-      PHPKeyword.VAR);
+  public void visitClassPropertyDeclaration(ClassPropertyDeclarationTree tree) {
+    SyntaxToken varToken = getVarModifier(tree);
+    if (varToken != null) {
+      context().newIssue(this.KEY, MESSAGE).tree(varToken);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Change this instance-reference to a static reference. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS2209)